File: 292_Nim_Game/cpp/solution.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool canWinNim(int n) 
    {
        return !(n%4 == 0);
    }
};

int main()
{
	int testCase[] = {0, 1, 2, 4, 8, 9, 112, 439, 23234};
	Solution s;

	for (int i=0; i<9; i++) 
	{
		cout << "[" << testCase[i] << "]: " << s.canWinNim(testCase[i]) << endl;
	}

	return 0;
}

File: 292_Nim_Game/java/Solution.java
public class Solution {
  public boolean canWinNim(int n) {
    return !(n % 4 == 0);
  }

  public static void main(String[] args) {

    int[] testCase = {1, 2, 3, 4, 9, 12};
    Solution s = new Solution();
    for (int i=0; i<testCase.length; i++) {
      System.out.println("[" + testCase[i] + "]:" + 
          s.canWinNim(testCase[i]));
    }
  }
}

